sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusRetryPolicy.cs

```diff
@@ -185,49 +192,42 @@ public abstract class ServiceBusRetryPolicy
             throw new TaskCanceledException();
         }
 
-        internal void SetServerBusy(string exceptionMessage)
+        private void SetServerBusy(string exceptionMessage, CancellationToken cancellationToken)
         {
             // multiple call to this method will not prolong the timer.
-            if (encounteredServerBusy)
+            if (serverBusyState == ServerBusyState)
             {
                 return;
             }
 
-            lock (serverBusyLock)
-            {
-                if (!encounteredServerBusy)
-                {
-                    encounteredServerBusy = true;
-                    ServerBusyExceptionMessage = string.IsNullOrWhiteSpace(exceptionMessage) ?
-                        Resources.DefaultServerBusyException : exceptionMessage;
-                    IsServerBusy = true;
-                    _ = ScheduleResetServerBusy();
-                }
-            }
+            ServerBusyExceptionMessage = string.IsNullOrWhiteSpace(exceptionMessage) ?
+                Resources.DefaultServerBusyException : exceptionMessage;
+            Interlocked.Exchange(ref serverBusyState, ServerBusyState);
+            _ = ScheduleResetServerBusy(cancellationToken);
         }
 
-        internal void ResetServerBusy()
+        private void ResetServerBusy()
         {
-            if (!encounteredServerBusy)
+            if (serverBusyState == ServerNotBusyState)
             {
                 return;
             }
 
-            lock (serverBusyLock)
-            {
-                if (encounteredServerBusy)
-                {
-                    encounteredServerBusy = false;
-                    ServerBusyExceptionMessage = Resources.DefaultServerBusyException;
-                    IsServerBusy = false;
-                }
-            }
+            ServerBusyExceptionMessage = Resources.DefaultServerBusyException;
```

Comment:
Aside from doing away with the lock, I think the code is a lot clearer now. Thanks @danielmarbach!

Reply:
Addressed the nits. I shied away from adding more tests that cover additional already previously untested branches of the policy for now, since I feel I have done enough "leave the camping ground cleaner than it was before :wastebasket: " for now. I hope you don't mind me being a bit lazy for this time ;)